aarch64 fails with newer glibc (ipc_perm.mode)

I'm playing around with some changes that make it easier to run the tests for multiple architectures locally based on recent work merged into Nix.  Running the build/tests in [posborne/rust-cross:arm](https://hub.docker.com/r/posborne/rust-cross/) which is based on Ubuntu 15.10, this following error is received:

```
/data/aarch64-unknown-linux-gnu/debug/build/libc-test-2ff1f50e28187d2d/out/all.c: In function '__test_field_type_ipc_perm_mode':
/data/aarch64-unknown-linux-gnu/debug/build/libc-test-2ff1f50e28187d2d/out/all.c:9148:21: error: return from incompatible pointer type [-Werror]
                     return &b->mode;
                     ^
cc1: all warnings being treated as errors
thread '<main>' panicked at 'explicit panic', /root/.multirust/toolchains/1.7.0/cargo/registry/src/github.com-88ac128001ac3a9a/gcc-0.3.25/src/lib.rs:818
```

It appears that the type of `struct ipc_perm.mode` has a differnt type (`mode_t` I think).  Not sure how best to handle.  Simply changing the type to `mode_t` causes other failures.

--- a/ci/docker/aarch64-unknown-linux-gnu/Dockerfile
+++ b/ci/docker/aarch64-unknown-linux-gnu/Dockerfile
@@ -1,4 +1,4 @@
-FROM ubuntu:14.04
+FROM ubuntu:16.04
 RUN apt-get update
 RUN apt-get install -y --no-install-recommends \
   gcc libc6-dev ca-certificates \

--- a/src/unix/notbsd/linux/other/b64/aarch64.rs
+++ b/src/unix/notbsd/linux/other/b64/aarch64.rs
@@ -60,10 +60,9 @@ s! {
         pub gid: ::gid_t,
         pub cuid: ::uid_t,
         pub cgid: ::gid_t,
-        pub mode: ::c_ushort,
-        __pad1: ::c_ushort,
+        pub mode: ::c_uint,
         pub __seq: ::c_ushort,
-        __pad2: ::c_ushort,
+        __pad1: ::c_ushort,
         __unused1: ::c_ulong,
         __unused2: ::c_ulong
     }
@@ -110,7 +109,7 @@ pub const SYS_perf_event_open: ::c_long = 241;
 pub const MCL_CURRENT: ::c_int = 0x0001;
 pub const MCL_FUTURE: ::c_int = 0x0002;
 
-pub const SIGSTKSZ: ::size_t = 8192;
+pub const SIGSTKSZ: ::size_t = 16384;
 pub const CBAUD: ::tcflag_t = 0o0010017;
 pub const TAB1: ::c_int = 0x00000800;
 pub const TAB2: ::c_int = 0x00001000;

